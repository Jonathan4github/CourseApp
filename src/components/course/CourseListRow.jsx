import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';


const propTypes = {
  course: PropTypes.shape().isRequired,
  handleDelete: PropTypes.func.isRequired
};

const CourseListRow = ({ course, handleDelete }) => (
  <tr>
    <td>
      <a href={course.watchHref} target="_blank" rel="noopener noreferrer">Watch</a>
    </td>
    <td><Link to={`/course/${course.id}`}>{course.title}</Link></td>
    <td>{course.authorId}</td>
    <td>{course.category}</td>
    <td>{course.length}</td>
    <td>
      <button type="button" onClick={() => handleDelete(course.id)} className="btn btn-danger">Delete</button>
    </td>
  </tr>
);

CourseListRow.propTypes = propTypes;

export default CourseListRow;
