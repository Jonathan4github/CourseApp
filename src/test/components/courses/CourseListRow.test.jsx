/* eslint-disable */
import CourseListRow from '../../../components/course/CourseListRow';
import coursesMock from '../../__mocks__/courseMock';


describe('CourseListRow component', () => {
  const setup = () => {
    const props = {
      course: coursesMock.existingCourse,
      handleDelete: jest.fn()
    };

    return {
      shallowWrapper: shallow(<CourseListRow {...props} />),
      props
    };
  };


  it('should render without crashing', () => {
    const { shallowWrapper } = setup();

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('should invoke the handleDelete method', () => {
    const { shallowWrapper, props } = setup();

    shallowWrapper.find('button').simulate('click', {});

    expect(props.handleDelete).toHaveBeenCalledWith(props.course.id);
  });
});
