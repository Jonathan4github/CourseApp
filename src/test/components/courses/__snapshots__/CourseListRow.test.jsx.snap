// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseListRow component should render without crashing 1`] = `
<tr>
  <td>
    <a
      className="btn btn-outline-secondary btn-sm"
      href="http://www.pluralsight.com/courses/react-flux-building-applications"
      rel="noopener noreferrer"
      target="_blank"
    >
      Watch
    </a>
  </td>
  <td>
    <Link
      replace={false}
      to="/courses/react-flux-building-applications"
    >
      Building Applications in React and Flux
    </Link>
  </td>
  <td>
    cory-house
  </td>
  <td>
    JavaScript
  </td>
  <td>
    5:08
  </td>
  <td>
    <button
      className="btn btn-outline-danger btn-sm"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </td>
</tr>
`;
